package-scripts/datadog-agent/postinst

```diff
@@ -28,41 +28,45 @@ if [ -f /etc/supervisor/conf.d/ddagent.conf ]; then
 fi
 
 if [ -f /etc/debian_version ] || [ "$LINUX_DISTRIBUTION" == "Debian" ] || [ "$LINUX_DISTRIBUTION" == "Ubuntu" ]; then
-  set -e
-  case "$1" in
-      configure)
-          update-rc.d datadog-agent defaults
-          adduser --system dd-agent --disabled-login --shell /bin/sh --no-create-home --quiet
-          usermod -d /opt/datadog-agent dd-agent
-          set +e
-      ;;
-      abort-upgrade|abort-remove|abort-deconfigure)
-      ;;
+    set -e
+    case "$1" in
+        configure)
+            update-rc.d datadog-agent defaults
+            update-rc.d datadog-agent enable 
+            adduser --system dd-agent --disabled-login --shell /bin/sh --no-create-home --quiet
+            usermod -d /opt/datadog-agent dd-agent
+            set +e
+        ;;
+        abort-upgrade|abort-remove|abort-deconfigure)
+        ;;
 
-      *)
-      ;;
-  esac
-  #DEBHELPER#
+        *)
+        ;;
+    esac
+    #DEBHELPER#
 fi
 
-chmod 755 /etc/init.d/datadog-agent
-
 # Set proper rights to the dd-agent user
 chown -R dd-agent:root ${CONFIG_DIR}
 chown -R dd-agent:root ${LOG_DIR}
 chown root:root /etc/init.d/datadog-agent
 chown -R root:root /opt/datadog-agent
 
-if which chkconfig >/dev/null 2>&1; then
-  chkconfig --add datadog-agent
+if command -v chkconfig >/dev/null 2>&1; then
+    chkconfig --add datadog-agent
 fi
 
-if which invoke-rc.d >/dev/null 2>&1; then
-    invoke-rc.d datadog-agent restart
-else
-    /etc/init.d/datadog-agent restart
-fi
+/etc/init.d/datadog-agent configcheck 
```

Comment:
Maybe add a comment here to explain the different status codes and why we skip restart in those cases?